roboteam_ai/src/coach/midField/HarassRobotCoach.cpp

```diff
@@ -0,0 +1,270 @@
+            if (distanceToHarasser < closestRobotToHarasserDistance) {
+                closestRobotToHarasserDistance = distanceToHarasser;
+                closestRobotToHarasser = std::make_shared<Robot>(robot);
+            }
+        }
+    }
+
+    if (goAfterBall) {
+        // see if opponent is too close to the ball
+        if (closestRobotToBallDistance <= TOO_CLOSE_TO_BALL_DISTANCE) {
+            // if this robot is closest to the robot with ball, harass that robot
+            int bestIndex = getRobotIndexCloseToEnemyRobot(closestRobotToBall);
+            if (bestIndex == myIndex) {
+                if (! robotAlreadyBeingHarassed(myIndex, closestRobotToBall->id)) {
+                    targetRobotsToHarass[myIndex] = closestRobotToBall;
+                    harassTarget.harassRobot = closestRobotToBall->id;
+                    harassTarget.harassPosition = harassRobot(myIndex, closestRobotToBall->id);
+                    return harassTarget;
+                }
+            }
+        }
+    }
+
+    // else, harass the opponent closest to the harasser
+
+
+    if(!robotAlreadyBeingHarassed(myIndex, closestRobotToHarasser->id)) {
+        targetRobotsToHarass[myIndex] = closestRobotToHarasser;
+        harassTarget.harassRobot = closestRobotToHarasser->id;
+        harassTarget.harassPosition = harassRobot(myIndex, closestRobotToHarasser->id);
+        return harassTarget;
+    }
+
+    // can't get a robot to harass, remove my targetRobot if it was there
+    targetRobotsToHarass[myIndex] = RobotPtr(nullptr);
+
+    harassTarget.harassRobot = -1;
+    harassTarget.harassPosition = standInMidField(thisRobot, myIndex);
+    return harassTarget;
+}
+
+Vector2 HarassRobotCoach::standInMidField(const HarassRobotCoach::RobotPtr &thisRobot, int &myIndex) {
+    Vector2 currentLocation = thisRobot->pos;
+
+    // if not in the right x-location yet, go there first
+    if (abs(currentLocation.x - bestXPos) > 0.5) {
+        Vector2 target = {bestXPos, currentLocation.y};
+        targetRobotPositions[myIndex] = target;
+        return target;
+    }
+
+    // find the best position to receive a pass
+    coach::PassScore passScore;
+    Vector2 bestPosition = thisRobot->pos;
+    double bestScore = passScore.calculatePassScore(bestPosition);
+    double currentX = thisRobot->pos.x;
+    double currentY = thisRobot->pos.y;
+    for (int yDiff = -GRID_SIZE; yDiff < GRID_SIZE; yDiff++) {
+        Vector2 newPos = {currentX, currentY + yDiff * GRID_INTERVAL};
+        double newScore = passScore.calculatePassScore(newPos);
+        if (newScore > bestScore) {
+            bestPosition = newPos;
+            bestScore = newScore;
+        }
+    }
+
+    targetRobotPositions[myIndex] = bestPosition;
+
+    // make sure you are not too close to the other midfielders and the side
+    for (int i = 0; i < static_cast<int>(currentRobotPositions.size()); i ++) {
+        if (i == myIndex) continue;
```

Comment:
erm... myindex?

Reply:
Yep, was already there and I just stuck with it

Reply:
hah that's me
it's beatiful and it works

Reply:
/sideeffects